### PR TITLE
Fix actuators

### PR DESCRIPTION
**Pull request by stimpfli**
### Description
The behaviour of the Arista was not symmetric. Removing the unused Arista actuators fixed the problem. As a consequence actuators were removed from the xmls and added in the code depending on the control type and only for the actuated joints.  The actuator kp and forecrange can know be either a list (for joint level intialization) or a float the would be applied to actuators. The actuator_gain can either be the kp for position actuators, the kv for velocity actuators and will be ignored for torque actuators. 

Add joint sensors had to be adjusted by removing the sensors for the unused actuators and only adding a single force sensor per joint that is linked to the actuator. 
Similarly get_observation was adjusted. 

### Types of changes

- [x] Bugfix
- [x] New feature
- [ ] Refactor / Code style update (no logical changes)
- [ ] Build / CI changes
- [ ] Documentation Update
- [ ] Other (explain)

### Does this address any currently open issues?
Partially #119 as it removes the actuators from the xml.

### CODE REVIEW

File: flygym/fly.py [hunk 1]
```diff
@@ -922,6 +887,64 @@ def _add_force_sensors(self):
...
+    def _add_joint_actuators(self, gain, forcerange):
+        # if self control is "motor" check that the gain is not provided
+        if self.control == "motor" and gain is not None:
+            # print warning
+            logging.warning(
+                "Motor control is selected, the gain parameter will not be used"
+            )
+
+        ## Need to deal with the kp, force range, and neck actuators
+        if not type(gain) == list:
```
Comment (sibocw) [inline flygym/fly.py:899]: In general type checking should be done with `isinstance(variable, type)`. Here: `if not isinstance(gain, list)`. Same below. See https://stackoverflow.com/a/73596548/13560623.

File: flygym/fly.py [hunk 2]
```diff
@@ -922,6 +887,64 @@ def _add_force_sensors(self):
...
+                actuators.append(
+                    self.model.actuator.add(
+                        self.control,
+                        name=f"actuator_{self.control}_{joint}",
+                        joint=joint,
+                        ctrlrange="-1000000 1000000",
+                        forcerange=forcerange,
+                        forcelimited=True,
+                    )
+                )
```
Comment (sibocw) [inline flygym/fly.py:944]: I suggest adding an else clause:
```
else:
    raise ValueError(f"Invalid control type {control}.")
```
If something like this already exists in `__init__`, I suggest removing it and putting it here instead.

File: flygym/fly.py [hunk 3]
```diff
@@ -821,18 +796,8 @@ def _add_joint_sensors(self):
...
-                        "actuatorfrc",
-                        name=f"actuatorfrc_velocity_{joint}",
-                        actuator=f"actuator_velocity_{joint}",
-                    ),
-                    self.model.sensor.add(
-                        "actuatorfrc",
-                        name=f"actuatorfrc_motor_{joint}",
-                        actuator=f"actuator_torque_{joint}",
+                        name=f"actuatorfrc_{self.control}_{joint}",
+                        actuator=f"actuator_{self.control}_{joint}",
```
Comment (sibocw) [inline flygym/fly.py:800]: Have you checked if this gives the same force readouts as before?

File: flygym/fly.py [hunk 4]
```diff
@@ -939,7 +962,7 @@ def _add_adhesion_actuators(self, gain):
 
     def set_pose(self, pose: state.KinematicPose, physics: mjcf.Physics):
         for i in range(len(self.actuated_joints)):
-            curr_joint = self.actuators[i].joint.name
+            curr_joint = self.actuators[i].joint
```
Comment (sibocw) [inline flygym/fly.py:965]: Why is `.name` no longer needed?

**Review by sibocw — COMMENTED** [review]
Thanks @stimpfli! I just have a few questions.
Did you check if the files under `flygym/examples` need to be changed as a result of this? Did we ever manually specify a gain or a controller type? If you haven't checked yet, I can do it also.

File: flygym/fly.py (hunk 4 above)
Comment (stimpfli) [reply flygym/fly.py:965]: For an unknown reason self.actuators[I].joints yields a joint object if obtained through the xml but only a string if the actuator is added in the code.

**Comment by stimpfli** [comment]
> under

I have just checked that we never mentioned actuator_kp.
For the control part I have not done it yet. I am pretty sure we never used the motor actuators right ?

**Review by sibocw — APPROVED** [review]
Thanks! Look good to me

**Pull request merged**